# 1. Business Understanding


# 2. Data Understanding

In [25]:
# importing data 
import pandas as pd
import numpy as np 

df = pd.read_csv('data.csv', sep=';')
df.head()

Marital status  Application mode  Application order  Course  \
0               1                17                  5     171   
1               1                15                  1    9254   
2               1                 1                  5    9070   
3               1                17                  2    9773   
4               2                39                  1    8014   

   Daytime/evening attendance\t  Previous qualification  \
0                             1                       1   
1                             1                       1   
2                             1                       1   
3                             1                       1   
4                             0                       1   

   Previous qualification (grade)  Nacionality  Mother's qualification  \
0                           122.0            1                      19   
1                           160.0            1                       1   
2                           122.0            1                      37   
3                           122.0            1                      38   
4                           100.0            1                      37   

   Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                      12  ...                                    0   
1                       3  ...                                    0   
2                      37  ...                                    0   
3                      37  ...                                    0   
4                      38  ...                                    0   

   Curricular units 2nd sem (enrolled)  \
0                                    0   
1                                    6   
2                                    6   
3                                    6   
4                                    6   

   Curricular units 2nd sem (evaluations)  \
0                                       0   
1                                       6   
2                                       0   
3                                      10   
4                                       6   

   Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
0                                    0                          0.000000   
1                                    6                         13.666667   
2                                    0                          0.000000   
3                                    5                         12.400000   
4                                    6                         13.000000   

   Curricular units 2nd sem (without evaluations)  Unemployment rate  \
0                                               0               10.8   
1                                               0               13.9   
2                                               0               10.8   
3                                               0                9.4   
4                                               0               13.9   

   Inflation rate   GDP    Target  
0             1.4  1.74   Dropout  
1            -0.3  0.79  Graduate  
2             1.4  1.74   Dropout  
3            -0.8 -3.12  Graduate  
4            -0.3  0.79  Graduate  

[5 rows x 37 columns]

### 2.1 - Victoria

### 2.2 - Anika

In [26]:
# Anika testing

# second test


In [27]:

pip install ucimlrepo

Note: you may need to restart the kernel to use updated packages.


DATA UNDERSTANDING

### A. Data set

In [28]:
df = pd.read_csv('data.csv', sep=';') 
df

Marital status  Application mode  Application order  Course  \
0                  1                17                  5     171   
1                  1                15                  1    9254   
2                  1                 1                  5    9070   
3                  1                17                  2    9773   
4                  2                39                  1    8014   
...              ...               ...                ...     ...   
4419               1                 1                  6    9773   
4420               1                 1                  2    9773   
4421               1                 1                  1    9500   
4422               1                 1                  1    9147   
4423               1                10                  1    9773   

      Daytime/evening attendance\t  Previous qualification  \
0                                1                       1   
1                                1                       1   
2                                1                       1   
3                                1                       1   
4                                0                       1   
...                            ...                     ...   
4419                             1                       1   
4420                             1                       1   
4421                             1                       1   
4422                             1                       1   
4423                             1                       1   

      Previous qualification (grade)  Nacionality  Mother's qualification  \
0                              122.0            1                      19   
1                              160.0            1                       1   
2                              122.0            1                      37   
3                              122.0            1                      38   
4                              100.0            1                      37   
...                              ...          ...                     ...   
4419                           125.0            1                       1   
4420                           120.0          105                       1   
4421                           154.0            1                      37   
4422                           180.0            1                      37   
4423                           152.0           22                      38   

      Father's qualification  ...  Curricular units 2nd sem (credited)  \
0                         12  ...                                    0   
1                          3  ...                                    0   
2                         37  ...                                    0   
3                         37  ...                                    0   
4                         38  ...                                    0   
...                      ...  ...                                  ...   
4419                       1  ...                                    0   
4420                       1  ...                                    0   
4421                      37  ...                                    0   
4422                      37  ...                                    0   
4423                      37  ...                                    0   

      Curricular units 2nd sem (enrolled)  \
0                                       0   
1                                       6   
2                                       6   
3                                       6   
4                                       6   
...                                   ...   
4419                                    6   
4420                                    6   
4421                                    8   
4422                                    5   
4423                                    6   

      Curricular units 2nd sem (evaluations)  \
0                            

In [29]:
from ucimlrepo import fetch_ucirepo 
  
# fetch dataset 
predict_students_dropout_and_academic_success = fetch_ucirepo(id=697) 
 
# variable information 
df_var = predict_students_dropout_and_academic_success.variables
df_var

name     role         type  \
0                                   Marital Status  Feature      Integer   
1                                 Application mode  Feature      Integer   
2                                Application order  Feature      Integer   
3                                           Course  Feature      Integer   
4                       Daytime/evening attendance  Feature      Integer   
5                           Previous qualification  Feature      Integer   
6                   Previous qualification (grade)  Feature   Continuous   
7                                      Nacionality  Feature      Integer   
8                           Mother's qualification  Feature      Integer   
9                           Father's qualification  Feature      Integer   
10                             Mother's occupation  Feature      Integer   
11                             Father's occupation  Feature      Integer   
12                                 Admission grade  Feature   Continuous   
13                                       Displaced  Feature      Integer   
14                       Educational special needs  Feature      Integer   
15                                          Debtor  Feature      Integer   
16                         Tuition fees up to date  Feature      Integer   
17                                          Gender  Feature      Integer   
18                              Scholarship holder  Feature      Integer   
19                               Age at enrollment  Feature      Integer   
20                                   International  Feature      Integer   
21             Curricular units 1st sem (credited)  Feature      Integer   
22             Curricular units 1st sem (enrolled)  Feature      Integer   
23          Curricular units 1st sem (evaluations)  Feature      Integer   
24             Curricular units 1st sem (approved)  Feature      Integer   
25                Curricular units 1st sem (grade)  Feature      Integer   
26  Curricular units 1st sem (without evaluations)  Feature      Integer   
27             Curricular units 2nd sem (credited)  Feature      Integer   
28             Curricular units 2nd sem (enrolled)  Feature      Integer   
29          Curricular units 2nd sem (evaluations)  Feature      Integer   
30             Curricular units 2nd sem (approved)  Feature      Integer   
31                Curricular units 2nd sem (grade)  Feature      Integer   
32  Curricular units 2nd sem (without evaluations)  Feature      Integer   
33                               Unemployment rate  Feature   Continuous   
34                                  Inflation rate  Feature   Continuous   
35                                             GDP  Feature   Continuous   
36                                          Target   Target  Categorical   

        demographic                                        description units  \
0    Marital Status  1 – single 2 – married 3 – widower 4 – divorce...  None   
1              None  1 - 1st phase - general contingent 2 - Ordinan...  None   
2              None  Application order (between 0 - first choice; a...  None   
3              None  33 - Biofuel Production Technologies 171 - Ani...  None   
4              None                            1 – daytime 0 - evening  None   
5   Education Level  1 - Secondary education 2 - Higher education -...  None   
6              None  Grade of previous qualification (between 0 and...  None   
7       Nationality  1 - Portuguese; 2 - German; 6 - Spanish; 11 - ...  None   
8   Education Level  1 - Secondary Education - 12th Year of Schooli...  None   
9   Education Level  1 - Secondary Education - 12th Year of Schooli...  None   
10       Occupation  0 - Student 1 - Representatives of the Legisla...  None   
11       Occupation  0 - Student 1 - Representatives of the Legisla...  None   
12             None                Admission grade (between 0 and 200)  None   
13             None                  

### B. Data Information and Data Types:

*The data information and data types from the original dataset are as listed above.*

For the most part, it would be best to represent each data attribute the way it's listed, but the 
following attributes would benefit from using a different data type:  
* Course: string instead of int  
* Nationality: string instead of int

For these variables, it would be best to work with them as integers, but they are thought of as booleans:
   * Displaced, Educational special needs, Debtor, Tuition fees up to date, Scholarship holder, International

The following data attributes will not contribute meaninfully to our analysis:
   * Application mode, Application order, Admission grade, Previous qualification (grade)

#### Variable Descriptions:

**Marital status:** 

- 1 – single 2 – married 3 – widower 4 – divorced 5 – facto union 6 – legally separated

**Application mode:** 

- 1 - 1st phase - general contingent 2 - Ordinance No. 612/93 5 - 1st phase - special contingent (Azores Island) 7 - Holders of other higher courses 10 - Ordinance No. 854-B/99 15 - International student (bachelor) 16 - 1st phase - special contingent (Madeira Island) 17 - 2nd phase - general contingent 18 - 3rd phase - general contingent 26 - Ordinance No. 533-A/99, item b2) (Different Plan) 27 - Ordinance No. 533-A/99, item b3 (Other Institution) 39 - Over 23 years old 42 - Transfer 43 - Change of course 44 - Technological specialization diploma holders 51 - Change of institution/course 53 - Short cycle diploma holders 57 - Change of institution/course (International)

**Application order:**

- Application order (between 0 - first choice; and 9 last choice)


**Course:** 

- 33 - Biofuel Production Technologies 171 - Animation and Multimedia Design 8014 - Social Service (evening attendance) 9003 - Agronomy 9070 - Communication Design 9085 - Veterinary Nursing 9119 - Informatics Engineering 9130 - Equinculture 9147 - Management 9238 - Social Service 9254 - Tourism 9500 - Nursing 9556 - Oral Hygiene 9670 - Advertising and Marketing Management 9773 - Journalism and Communication 9853 - Basic Education 9991 - Management (evening attendance)

**Daytime/evening attendance:** 

- 1 – daytime 0 - evening

**Previous qualification:**

 - 1 - Secondary education 2 - Higher education - bachelor's degree 3 - Higher education - degree 4 - Higher education - master's 5 - Higher education - doctorate 6 - Frequency of higher education 9 - 12th year of schooling - not completed 10 - 11th year of schooling - not completed 12 - Other - 11th year of schooling 14 - 10th year of schooling 15 - 10th year of schooling - not completed 19 - Basic education 3rd cycle (9th/10th/11th year) or equiv. 38 - Basic education 2nd cycle (6th/7th/8th year) or equiv. 39 - Technological specialization course 40 - Higher education - degree (1st cycle) 42 - Professional higher technical course 43 - Higher education - master (2nd cycle)

**Previous qualification (grade):** 

- Grade of previous qualification (between 0 and 200)

**Nationality:** 

- 1 - Portuguese; 2 - German; 6 - Spanish; 11 - Italian; 13 - Dutch; 14 - English; 17 - Lithuanian; 21 - Angolan; 22 - Cape Verdean; 24 - Guinean; 25 - Mozambican; 26 - Santomean; 32 - Turkish; 41 - Brazilian; 62 - Romanian; 100 - Moldova (Republic of); 101 - Mexican; 103 - Ukrainian; 105 - Russian; 108 - Cuban; 109 - Colombian

**Mother's qualification:**

 - 1 - Secondary Education - 12th Year of Schooling or Eq. 2 - Higher Education - Bachelor's Degree 3 - Higher Education - Degree 4 - Higher Education - Master's 5 - Higher Education - Doctorate 6 - Frequency of Higher Education 9 - 12th Year of Schooling - Not Completed 10 - 11th Year of Schooling - Not Completed 11 - 7th Year (Old) 12 - Other - 11th Year of Schooling 14 - 10th Year of Schooling 18 - General commerce course 19 - Basic Education 3rd Cycle (9th/10th/11th Year) or Equiv. 22 - Technical-professional course 26 - 7th year of schooling 27 - 2nd cycle of the general high school course 29 - 9th Year of Schooling - Not Completed 30 - 8th year of schooling 34 - Unknown 35 - Can't read or write 36 - Can read without having a 4th year of schooling 37 - Basic education 1st cycle (4th/5th year) or equiv. 38 - Basic Education 2nd Cycle (6th/7th/8th Year) or Equiv. 39 - Technological specialization course 40 - Higher education - degree (1st cycle) 41 - Specialized higher studies course 42 - Professional higher technical course 43 - Higher Education - Master (2nd cycle) 44 - Higher Education - Doctorate (3rd cycle)

**Father's qualification:**

 - 1 - Secondary Education - 12th Year of Schooling or Eq. 2 - Higher Education - Bachelor's Degree 3 - Higher Education - Degree 4 - Higher Education - Master's 5 - Higher Education - Doctorate 6 - Frequency of Higher Education 9 - 12th Year of Schooling - Not Completed 10 - 11th Year of Schooling - Not Completed 11 - 7th Year (Old) 12 - Other - 11th Year of Schooling 14 - 10th Year of Schooling 18 - General commerce course 19 - Basic Education 3rd Cycle (9th/10th/11th Year) or Equiv. 22 - Technical-professional course 26 - 7th year of schooling 27 - 2nd cycle of the general high school course 29 - 9th Year of Schooling - Not Completed 30 - 8th year of schooling 34 - Unknown 35 - Can't read or write 36 - Can read without having a 4th year of schooling 37 - Basic education 1st cycle (4th/5th year) or equiv. 38 - Basic Education 2nd Cycle (6th/7th/8th Year) or Equiv. 39 - Technological specialization course 40 - Higher education - degree (1st cycle) 41 - Specialized higher studies course 42 - Professional higher technical course 43 - Higher Education - Master (2nd cycle) 44 - Higher Education - Doctorate (3rd cycle)

**Mother's occupation:**

 - 0 - Student 1 - Representatives of the Legislative Power and Executive Bodies, Directors, Directors and Executive Managers 2 - Specialists in Intellectual and Scientific Activities 3 - Intermediate Level Technicians and Professions 4 - Administrative staff 5 - Personal Services, Security and Safety Workers and Sellers 6 - Farmers and Skilled Workers in Agriculture, Fisheries and Forestry 7 - Skilled Workers in Industry, Construction and Craftsmen 8 - Installation and Machine Operators and Assembly Workers 9 - Unskilled Workers 10 - Armed Forces Professions 90 - Other Situation 99 - (blank) 122 - Health professionals 123 - teachers 125 - Specialists in information and communication technologies (ICT) 131 - Intermediate level science and engineering technicians and professions 132 - Technicians and professionals, of intermediate level of health 134 - Intermediate level technicians from legal, social, sports, cultural and similar services 141 - Office workers, secretaries in general and data processing operators 143 - Data, accounting, statistical, financial services and registry-related operators 144 - Other administrative support staff 151 - personal service workers 152 - sellers 153 - Personal care workers and the like 171 - Skilled construction workers and the like, except electricians 173 - Skilled workers in printing, precision instrument manufacturing, jewelers, artisans and the like 175 - Workers in food processing, woodworking, clothing and other industries and crafts 191 - cleaning workers 192 - Unskilled workers in agriculture, animal production, fisheries and forestry 193 - Unskilled workers in extractive industry, construction, manufacturing and transport 194 - Meal preparation assistants

**Father's occupation:**

 - 0 - Student 1 - Representatives of the Legislative Power and Executive Bodies, Directors, Directors and Executive Managers 2 - Specialists in Intellectual and Scientific Activities 3 - Intermediate Level Technicians and Professions 4 - Administrative staff 5 - Personal Services, Security and Safety Workers and Sellers 6 - Farmers and Skilled Workers in Agriculture, Fisheries and Forestry 7 - Skilled Workers in Industry, Construction and Craftsmen 8 - Installation and Machine Operators and Assembly Workers 9 - Unskilled Workers 10 - Armed Forces Professions 90 - Other Situation 99 - (blank) 122 - Health professionals 123 - teachers 125 - Specialists in information and communication technologies (ICT) 131 - Intermediate level science and engineering technicians and professions 132 - Technicians and professionals, of intermediate level of health 134 - Intermediate level technicians from legal, social, sports, cultural and similar services 141 - Office workers, secretaries in general and data processing operators 143 - Data, accounting, statistical, financial services and registry-related operators 144 - Other administrative support staff 151 - personal service workers 152 - sellers 153 - Personal care workers and the like 171 - Skilled construction workers and the like, except electricians 173 - Skilled workers in printing, precision instrument manufacturing, jewelers, artisans and the like 175 - Workers in food processing, woodworking, clothing and other industries and crafts 191 - cleaning workers 192 - Unskilled workers in agriculture, animal production, fisheries and forestry 193 - Unskilled workers in extractive industry, construction, manufacturing and transport 194 - Meal preparation assistants

**Admission grade:** Admission grade (between 0 and 200)

**Displaced:** 1 – yes 0 – no

**Educational special needs:** 1 – yes 0 – no

**Debtor:** 1 – yes 0 – no

**Tuition fees up to date:** 1 – yes 0 – no

**Gender:** 1 – male 0 – female

**Scholarship holder:** 1 – yes 0 – no

**Age at enrollment:** Age of studend at enrollment

**International:** 1 – yes 0 – no

**Curricular units 1st sem (credited):** Number of curricular units credited in the 1st semester

**Curricular units 1st sem (enrolled):** Number of curricular units enrolled in the 1st semester

**Curricular units 1st sem (evaluations):** Number of evaluations to curricular units in the 1st semester

**Curricular units 1st sem (approved):** Number of curricular units approved in the 1st semester	

**Curricular units 1st sem (grade):** Grade average in the 1st semester (between 0 and 20)

**Curricular units 1st sem (without evaluations):** Number of curricular units without evalutions in the 1st semester

**Curricular units 2nd sem (credited):** Number of curricular units credited in the 2nd semester	

**Curricular units 2nd sem (enrolled):** Number of curricular units enrolled in the 2nd semester	

**Curricular units 2nd sem (evaluations):** Number of evaluations to curricular units in the 2nd semester

**Curricular units 2nd sem (approved):** Number of curricular units approved in the 2nd semester	

**Curricular units 2nd sem (grade):** Grade average in the 2nd semester (between 0 and 20)	

**Curricular units 2nd sem (without evaluations):** Number of curricular units without evalutions in the 1st semester

**Unemployment rate:** Unemployment rate (%)

**Inflation rate:** Inflation rate (%)

**GDP:** GDP

**Target:** Target. The problem is formulated as a three category classification task (dropout, enrolled, and graduate) at the end of the normal duration of the courseing

#### Data quality

In [30]:
# Check duplicates

idx = df.duplicated()
len(df[idx])

0

There are no missing values in this dataset, as indicated by the data editor on dataset information page 
    from the UCI Irvine website
    [INSERT WHAT WE WOULD DO IF THERE WERE MISSING DATA]

There are no duplicate data values in this dataset
    [INSERT WHAT WE WOULD DO IF THERE WERE DUPLICATES]

This dataset has also been processed to remove outliers.  If there were outliers, we would drop all rows with entires
    that are IQR * 1.5 above Q3 and IQR * 1.5 below Q1 for that attribute.

#### Cleaning up data

In [31]:
# Replacing all course numbers with course names because the numbers aren't hierarchical, and the names
#    are more meaningful, as we can analyze the difficulty of the concepts

df["Course"].replace(to_replace = [33, 171, 8014, 9003, 9070, 9085, 9119, 9130, 9147, 9238, 9254, 9500, 9556, 9670, 9773, 9853, 9991],
                    value = ["Biofuel production", "Animation and multimedia", "Social service (evening)", "Agronomy", "Communication design", "Veterinary nursing", 
                            "Informatics engineering", "Equinculture", "Management", "Social service", "Tourism", "Nursing", "Oral hygiene", "Advertising and marketing management",
                            "Journalism and communication", "Basic Education", "Management (evening)"], inplace = True)

df["Course"]

0           Animation and multimedia
1                            Tourism
2               Communication design
3       Journalism and communication
4           Social service (evening)
                    ...             
4419    Journalism and communication
4420    Journalism and communication
4421                         Nursing
4422                      Management
4423    Journalism and communication
Name: Course, Length: 4424, dtype: object

In [32]:
# Correct the mispelling of "Nationality"

# Replacing all nationality numers with nationality names because the numbers aren't hierarchical, and the names
#    are more meaningful, as we can analyze where students come from

df.rename(columns = {"Nacionality":"Nationality"}, inplace = True)

df["Nationality"].replace(to_replace = [1, 2, 6, 11, 13, 14, 17, 21, 22, 24, 25, 26, 32, 41, 62, 100, 101, 103, 105, 108, 109],
                         value = ["Portuguese", "German", "Spanish", "Italian", "Dutch", "English", "Lithuanian", "Angolan", "Cape Verdean", 
                                  "Guinean", "Mozambican", "Santomean", "Turkish", "Brazillian", "Romanian", "Rep. of Moldova", "Mexican", "Ukranian", 
                                  "Russian", "Cuban", "Colombian"], inplace = True)
df["Nationality"]

0         Portuguese
1         Portuguese
2         Portuguese
3         Portuguese
4         Portuguese
            ...     
4419      Portuguese
4420         Russian
4421      Portuguese
4422      Portuguese
4423    Cape Verdean
Name: Nationality, Length: 4424, dtype: object

In [33]:
# Make casing of variable name consistent with others

# Replacing all marital status numbers with marital status names because the numbers aren't hierarchical, and the names
#    are more meaningful, as we can analyze a student's marital status categorically

df.rename(columns = {"Marital Status":"Marital status"}, inplace = True)

df["Marital status"].replace(to_replace = [1, 2, 3, 4, 5, 6],
                         value = ["single", "married", "widower", "divorced", "facto union", "legally separated"], inplace = True)

In [34]:
# Eliminated all rows with students "Enrolled" in order to focus on outcomes ("Graduates" and "Dropouts")

df = df.drop(df[df["Target"] == "Enrolled"].index)
df

Marital status  Application mode  Application order  \
0            single                17                  5   
1            single                15                  1   
2            single                 1                  5   
3            single                17                  2   
4           married                39                  1   
...             ...               ...                ...   
4419         single                 1                  6   
4420         single                 1                  2   
4421         single                 1                  1   
4422         single                 1                  1   
4423         single                10                  1   

                            Course  Daytime/evening attendance\t  \
0         Animation and multimedia                             1   
1                          Tourism                             1   
2             Communication design                             1   
3     Journalism and communication                             1   
4         Social service (evening)                             0   
...                            ...                           ...   
4419  Journalism and communication                             1   
4420  Journalism and communication                             1   
4421                       Nursing                             1   
4422                    Management                             1   
4423  Journalism and communication                             1   

      Previous qualification  Previous qualification (grade)   Nationality  \
0                          1                           122.0    Portuguese   
1                          1                           160.0    Portuguese   
2                          1                           122.0    Portuguese   
3                          1                           122.0    Portuguese   
4                          1                           100.0    Portuguese   
...                      ...                             ...           ...   
4419                       1                           125.0    Portuguese   
4420                       1                           120.0       Russian   
4421                       1                           154.0    Portuguese   
4422                       1                           180.0    Portuguese   
4423                       1                           152.0  Cape Verdean   

      Mother's qualification  Father's qualification  ...  \
0                         19                      12  ...   
1                          1                       3  ...   
2                         37                      37  ...   
3                         38                      37  ...   
4                         37                      38  ...   
...                      ...                     ...  ...   
4419                       1                       1  ...   
4420                       1                       1  ...   
4421                      37                      37  ...   
4422                      37                      37  ...   
4423                      38                      37  ...   

      Curricular units 2nd sem (credited)  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   
...                                   ...   
4419                                    0   
4420                                    0   
4421                                    0   
4422                                    0   
4423                                    0   

      Curricular units 2nd sem (enrolled)  \
0                                       0   
1                                       6   
2                                       6   
3                                       6   
4                              

In [35]:
# Eliminate attributes that are not necessary

# The following data attributes will not contribute meaninfully to our analysis:
#    Application mode, Application order, Previous qualification (grade), Admission grade

df = df.drop("Application mode", axis = 1)
df = df.drop("Application order", axis = 1)
df = df.drop("Previous qualification (grade)", axis = 1)
df = df.drop("Admission grade", axis = 1)

df

Marital status                        Course  \
0            single      Animation and multimedia   
1            single                       Tourism   
2            single          Communication design   
3            single  Journalism and communication   
4           married      Social service (evening)   
...             ...                           ...   
4419         single  Journalism and communication   
4420         single  Journalism and communication   
4421         single                       Nursing   
4422         single                    Management   
4423         single  Journalism and communication   

      Daytime/evening attendance\t  Previous qualification   Nationality  \
0                                1                       1    Portuguese   
1                                1                       1    Portuguese   
2                                1                       1    Portuguese   
3                                1                       1    Portuguese   
4                                0                       1    Portuguese   
...                            ...                     ...           ...   
4419                             1                       1    Portuguese   
4420                             1                       1       Russian   
4421                             1                       1    Portuguese   
4422                             1                       1    Portuguese   
4423                             1                       1  Cape Verdean   

      Mother's qualification  Father's qualification  Mother's occupation  \
0                         19                      12                    5   
1                          1                       3                    3   
2                         37                      37                    9   
3                         38                      37                    5   
4                         37                      38                    9   
...                      ...                     ...                  ...   
4419                       1                       1                    5   
4420                       1                       1                    9   
4421                      37                      37                    9   
4422                      37                      37                    7   
4423                      38                      37                    5   

      Father's occupation  Displaced  ...  \
0                       9          1  ...   
1                       3          1  ...   
2                       9          1  ...   
3                       3          1  ...   
4                       9          0  ...   
...                   ...        ...  ...   
4419                    4          0  ...   
4420                    9          1  ...   
4421                    9          1  ...   
4422                    4          1  ...   
4423                    9          1  ...   

      Curricular units 2nd sem (credited)  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   
...                                   ...   
4419                                    0   
4420                                    0   
4421                                    0   
4422                                    0   
4423                                    0   

      Curricular units 2nd sem (enrolled)  \
0                                       0   
1                                       6   
2                                       6   
3                                       6   
4                                       6   
...                                   ...   
4419                                    6   
4420                                    6   
4421                                    8  

Numerical data summary of all variables with numerical values (categorical variables excluded)

In [36]:
# Data summary: 18 attributes displayed

summary1 = df.iloc[0:, 0:20]
summary1.describe()

Daytime/evening attendance\t  Previous qualification  \
count                   3630.000000             3630.000000   
mean                       0.887603                4.532231   
std                        0.315897               10.024134   
min                        0.000000                1.000000   
25%                        1.000000                1.000000   
50%                        1.000000                1.000000   
75%                        1.000000                1.000000   
max                        1.000000               43.000000   

       Mother's qualification  Father's qualification  Mother's occupation  \
count             3630.000000             3630.000000          3630.000000   
mean                19.986226               22.571625            10.138567   
std                 15.585278               15.275453            23.315697   
min                  1.000000                1.000000             0.000000   
25%                  2.000000                3.000000             4.000000   
50%                 19.000000               19.000000             5.000000   
75%                 37.000000               37.000000             9.000000   
max                 44.000000               44.000000           194.000000   

       Father's occupation    Displaced  Educational special needs  \
count           3630.00000  3630.000000                3630.000000   
mean              10.28292     0.549036                   0.011019   
std               22.40269     0.497658                   0.104407   
min                0.00000     0.000000                   0.000000   
25%                4.00000     0.000000                   0.000000   
50%                7.00000     1.000000                   0.000000   
75%                9.00000     1.000000                   0.000000   
max              195.00000     1.000000                   1.000000   

            Debtor  Tuition fees up to date       Gender  Scholarship holder  \
count  3630.000000              3630.000000  3630.000000         3630.000000   
mean      0.113774                 0.866116     0.344077            0.266942   
std       0.317580                 0.340575     0.475132            0.442423   
min       0.000000                 0.000000     0.000000            0.000000   
25%       0.000000                 1.000000     0.000000            0.000000   
50%       0.000000                 1.000000     0.000000            0.000000   
75%       0.000000                 1.000000     1.000000            1.000000   
max       1.000000                 1.000000     1.000000            1.000000   

       Age at enrollment  International  Curricular units 1st sem (credited)  \
count        3630.000000    3630.000000                          3630.000000   
mean           23.461157       0.023691                             0.754270   
std             7.827994       0.152107                             2.477277   
min            17.000000       0.000000                             0.000000   
25%            19.000000       0.000000                             0.000000   
50%            20.000000       0.000000                             0.000000   
75%            25.000000       0.000000                             0.000000   
max            70.000000       1.000000                            20.000000   

       Curricular units 1st sem (enrolled)  \
count                          3630.000000   
mean                              6.337466   
std                               2.570773   
min                               0.000000   
25%                               5.000000   
50%                               6.000000   
75%                               7.000000   
max                              26.000000   

       Curricular units 1st sem (evaluations)  
count                             3630.000000  
mean                                 8.071074  
std                                  4.286632  
min                                  0.000000  
25%     

In [37]:
# Data summary: 16 attributes displayed

summary2 = df.iloc[0:, 20:38]
summary2.describe()

Curricular units 1st sem (approved)  Curricular units 1st sem (grade)  \
count                          3630.000000                       3630.000000   
mean                              4.791460                         10.534860   
std                               3.237845                          5.057694   
min                               0.000000                          0.000000   
25%                               3.000000                         11.000000   
50%                               5.000000                         12.341429   
75%                               6.000000                         13.500000   
max                              26.000000                         18.875000   

       Curricular units 1st sem (without evaluations)  \
count                                     3630.000000   
mean                                         0.128926   
std                                          0.679111   
min                                          0.000000   
25%                                          0.000000   
50%                                          0.000000   
75%                                          0.000000   
max                                         12.000000   

       Curricular units 2nd sem (credited)  \
count                          3630.000000   
mean                              0.581818   
std                               2.022688   
min                               0.000000   
25%                               0.000000   
50%                               0.000000   
75%                               0.000000   
max                              19.000000   

       Curricular units 2nd sem (enrolled)  \
count                          3630.000000   
mean                              6.296419   
std                               2.263020   
min                               0.000000   
25%                               5.000000   
50%                               6.000000   
75%                               7.000000   
max                              23.000000   

       Curricular units 2nd sem (evaluations)  \
count                             3630.000000   
mean                                 7.763085   
std                                  3.964163   
min                                  0.000000   
25%                                  6.000000   
50%                                  8.000000   
75%                                 10.000000   
max                                 33.000000   

       Curricular units 2nd sem (approved)  Curricular units 2nd sem (grade)  \
count                          3630.000000                       3630.000000   
mean                              4.518457                         10.036155   
std                               3.162376                          5.481742   
min                               0.000000                          0.000000   
25%                               2.000000                         10.517857   
50%                               5.000000                         12.333333   
75%                               6.000000                         13.500000   
max                              20.000000                         18.571429   

       Curricular units 2nd sem (without evaluations)  Unemployment rate  \
count                                     3630.000000        3630.000000   
mean                                         0.142149          11.630358   
std                                          0.747670           2.667652   
min                                          0.000000           7.600000   
25%                                          0.000000           9.400000   
50%                                          0.000000          11.100000   
75%                                          0.000000          13.900000   
max                                         12.000000          16.200000   

       Inflation rate          GDP  
count     3630.000000  3630.000000  
mean         1.23159

Data summary needed to be expressed this way because all the attributes wouldn't fit in one display ("..." in the middle hid values)

DATA VISUALIZATIONS

In [38]:
grad = 0
drop = 0
none = 0

for x in df["Target"]:
    if x == "Graduate":
        grad += 1
    elif x == "Dropout":
        drop += 1
    else:
        none += 1
        
print("Number of graduates: ", grad)
print("Number of dropouts: ", drop)
print("Other: ", none, " (checking for extraneous data, there is none)")
print("Total number of students: ", (grad + drop + none), "(total matches number of rows)")


Number of graduates:  2209
Number of dropouts:  1421
Other:  0  (checking for extraneous data, there is none)
Total number of students:  3630 (total matches number of rows)


In [39]:
df_target_labs = ["Graduates", "Dropouts"]
df_target_vals = [grad, drop]

plt.bar(df_target_labs, df_target_vals)
plt.xlabel("Student Outcome")
plt.ylabel("Number of Students")
plt.title("Number of Students who Graduated or Dropped Out")
plt.show()

NameError: name 'plt' is not defined

In [ ]:
# Histogram: qualifications

# fig, axs = plt.subplots(nrows=2, ncols=1, figsize=(15, 15), sharey=True)

plt.boxplot(df["Mother's qualification"], vert = False)
plt.title("Distribution of Mother's qualification", fontsize = 13)
plt.xlabel("Education level (scale of 1 - 44)", fontsize = 12)
plt.show()

plt.boxplot(df["Father's qualification"], vert = False)
plt.title("Distribution of Father's qualification", fontsize = 13)
plt.xlabel("Education level (scale of 1 - 44)", fontsize = 12)
plt.show()


### 2.3 - Ayan

## 3. Data Visualization